engine/verification/utils/unittest/fixture.go

```diff
@@ -294,24 +288,28 @@ func ExecutionResultFixture(t *testing.T, chunkCount int, chain flow.Chain, refB
 		for i := range computationResult.StateCommitments {
 			endState := computationResult.StateCommitments[i]
 
-			var collectionID flow.Identifier
-
-			txNumber := 1 //default for system chunk
-			// account for system chunk being last
+			// generates chunk and chunk data pack
+			var chunkDataPack *flow.ChunkDataPack
+			var chunk *flow.Chunk
 			if i < len(computationResult.StateCommitments)-1 {
+				// generates chunk data pack fixture for non-system chunk
 				collectionGuarantee := executableBlock.Block.Payload.Guarantees[i]
 				completeCollection := executableBlock.CompleteCollections[collectionGuarantee.ID()]
-				collectionID = completeCollection.Collection().ID()
-				txNumber = len(completeCollection.Transactions)
-			} else {
-				collectionID = flow.ZeroID
-			}
+				collection := completeCollection.Collection()
 
-			eventsHash, err := flow.EventsListHash(computationResult.Events[i])
-			require.NoError(t, err)
+				eventsHash, err := flow.EventsListHash(computationResult.Events[i])
+				require.NoError(t, err)
+
+				chunk = execution.GenerateChunk(i, startState, endState, executableBlock.ID(), eventsHash, uint16(len(completeCollection.Transactions)))
+				chunkDataPack = execution.GenerateChunkDataPack(chunk.ID(), chunk.StartState, &collection, computationResult.Proofs[i])
+			} else {
+				// generates chunk data pack fixture for system chunk
+				eventsHash, err := flow.EventsListHash(computationResult.Events[i])
+				require.NoError(t, err)
 
-			chunk := execution.GenerateChunk(i, startState, endState, collectionID, executableBlock.ID(), eventsHash, uint16(txNumber))
-			chunkDataPack := execution.GenerateChunkDataPack(chunk, collectionID, computationResult.Proofs[i])
+				chunk = execution.GenerateChunk(i, startState, endState, executableBlock.ID(), eventsHash, uint16(1))
+				chunkDataPack = execution.GenerateChunkDataPack(chunk.ID(), chunk.StartState, nil, computationResult.Proofs[i])
+			}
```

Comment:
Decoupling chunk data pack for system chunks from non-system chunks for better extensibility and maintainability.